### PR TITLE
Support disabling Auth0

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR allows running the datatools UI without having to log in using Auth0.
When Auth0 is disabled:
- A single admin user is created and is the only user available.
- No user management, account management or feed activity watching are available.

The PR is based on https://github.com/ibi-group/datatools-ui/tree/auth-disable but simplifies the permission checks by creating an admin user on loading.

Rough edges caused by not knowing what panes will be available until after a few render passes - I think that can be addressed in a subsequent PR:
- The users page in the Admin section is blank.
- The profile page in the Account section is blank.

To disable Auth0, insert a line with `DISABLE_AUTH: true` in `env.yml` for both frontend and backend configurations.
No other changes are needed on the backend.

This PR *might* address #962.

### CODE REVIEW

**Review by miles-grant-ibigroup — APPROVED** [review]
Thanks for cleaning this up! Much better approach than mine. I've tested thoroughly and everything works fantastic! Excited for what this will allow for the project

File: lib/common/containers/LocalUserRetriever.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        'projects': [],
+        'client_id': AUTH0_CLIENT_ID,
+        'subscriptions': []
+      }
+    ],
+    'roles': [
+      'user'
+    ]
+  },
+  // FIXME: pick a better email address for both backend and frontend.
```
Comment (amy-corson-ibigroup) [inline lib/common/containers/LocalUserRetriever.js:31]: Are you looking for a real email address here or just a different fake/placeholder one?

File: lib/public/components/UserAccount.js
Comment (amy-corson-ibigroup) [inline lib/public/components/UserAccount.js:1]: Would it be worth it to add a status message in this pane if `AUTH0_DISABLED` is true, that says something along the lines of "You're running datatools without user authentication enabled. Features such as user management, account management and feed activity watching are unavailable"

**Review by amy-corson-ibigroup — APPROVED** [review]
Very cool fix! Works great for me locally. Just a couple questions.

File: lib/public/components/UserAccount.js
Comment (binh-dam-ibigroup) [reply lib/public/components/UserAccount.js:1]: Thoughts @miles-grant-ibigroup?

Comment (miles-grant-ibigroup) [reply lib/public/components/UserAccount.js:1]: The homepage might be a better place? Could add a warning above the recent activity section?

File: lib/common/containers/LocalUserRetriever.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/common/containers/LocalUserRetriever.js:31]: Probably a different one. This email address appears in the Notes section of each feed.

File: lib/public/components/UserAccount.js
Comment (binh-dam-ibigroup) [reply lib/public/components/UserAccount.js:1]: I added the message to the home page, above Recent Activity (https://github.com/ibi-group/datatools-ui/pull/967/commits/969af57cd614a2fefb6e2e6dcfd70bff9c7ddeba)

**Pull request merged**